Source/WebCore/svg/SVGImageElement.cpp

```diff
@@ -86,8 +86,17 @@ bool SVGImageElement::renderingTaintsOrigin() const
     if (!cachedImage)
         return false;
 
-    auto* image = cachedImage->image();
-    return image && image->renderingTaintsOrigin();
+    RefPtr image = cachedImage->image();
+    if (!image)
+        return false;
+
+    if (image->renderingTaintsOrigin())
+        return true;
+
+    if (image->sourceURL().protocolIsData())
+        return false;
+
+    return cachedImage->isCORSCrossOrigin();
```

Comment:
This part is almost the same the code in `HTMLImageElement::originClean`. Can we have a helper function in `CachedImage` so code is not duplicated?

Reply:
The issue is that there is a slight difference in when protocolIsData is checked: we check in first for SVG and last for images.

We should probably align with the SVG one, but it might be better to do it in a follow-up.
I'll file a bug to follow-up on this.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=258065